cooperative-map-cache-source-test/src/main/java/com/hazelcast/jet/tests/cooperativesource/CooperativeMapCacheSourceTest.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.cooperativesource;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.pipeline.BatchSource;
+import com.hazelcast.jet.pipeline.GenericPredicates;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.projection.Projections;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import javax.cache.Cache;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+public class CooperativeMapCacheSourceTest extends AbstractSoakTest {
+
+    private static final String TEST_PREFIX = CooperativeMapCacheSourceTest.class.getSimpleName();
+    private static final String SOURCE_MAP = TEST_PREFIX + "_SourceMap";
+    private static final String SINK_LOCAL_MAP = TEST_PREFIX + "_SinkLocalMap";
+    private static final String SINK_REMOTE_MAP = TEST_PREFIX + "_SinkRemoteMap";
+    private static final String SINK_LOCAL_CACHE = TEST_PREFIX + "_SinkLocalCache";
+    private static final String SINK_REMOTE_CACHE = TEST_PREFIX + "_SinkRemoteCache";
+    private static final String SINK_QUERY_LOCAL_MAP = TEST_PREFIX + "_SinkQueryLocalMap";
+    private static final String SINK_QUERY_REMOTE_MAP = TEST_PREFIX + "_SinkQueryRemoteMap";
+    private static final String SOURCE_CACHE = TEST_PREFIX + "_SourceCache";
+    private static final int SOURCE_MAP_ITEMS = 1_000_000;
```

Comment:
I think 1M is too much data, we'll have 6M data in the cluster after the jobs complete

Reply:
yes let's go with 100K and see how the cluster performs